script/release/notes/notes.js

```diff
@@ -497,8 +497,20 @@ const removeSupercededStackUpdates = (commits) => {
       continue;
     }
     const [, dep, version] = match;
-    if (!newest[dep] || newest[dep].version < version) {
+    if (!newest[dep]) {
       newest[dep] = { commit, version };
+      continue;
+    }
+
+    const newestSplit = newest[dep].version.split('.');
+    const replacementSplit = version.split('.');
```

Comment:
```suggestion
    const replacementSplit = version.split('.');
    if (newestSplit.length !== replacementSplit.length) {
      throw new Error(`Expected version strings to have same number of sections: ${newest[dep].version} and ${version}`);
    }
```

Might help catch malformed lines or changes in version string formats.